packages/editor/src/components/page-attributes/parent.js

```diff
@@ -1,44 +1,195 @@
 /**
  * External dependencies
  */
-import { get } from 'lodash';
+import { get, debounce } from 'lodash';
+import Autocomplete from 'accessible-autocomplete/react';
 
 /**
  * WordPress dependencies
  */
-import { __ } from '@wordpress/i18n';
+import { sprintf, __, _n } from '@wordpress/i18n';
 import { TreeSelect } from '@wordpress/components';
 import { compose } from '@wordpress/compose';
 import { withSelect, withDispatch } from '@wordpress/data';
+import { Component } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
 
 /**
  * Internal dependencies
  */
 import { buildTermsTree } from '../../utils/terms';
 
-export function PageAttributesParent( { parent, postType, items, onUpdateParent } ) {
-	const isHierarchical = get( postType, [ 'hierarchical' ], false );
-	const parentPageLabel = get( postType, [ 'labels', 'parent_item_colon' ] );
-	const pageItems = items || [];
-	if ( ! isHierarchical || ! parentPageLabel || ! pageItems.length ) {
-		return null;
+export class PageAttributesParent extends Component {
```

Comment:
Could we extract a general autocomplete/combobox component for the components package out of this?

Reply:
yes, my goal is extract this as a reusable component.